### PR TITLE
Add API usage constraints

### PR DESCRIPTION
**Pull request by teoEnache**
Added the rate limiting in the API reference, because we decided it should be mentioned there.

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 openapi: 3.0.0
 info:
   title: UP42 API
-  version: '1.0'
+  version: '1.1'
```
Comment (dc-up42) [inline openapi.yaml:4]: What is our approach to versioning in this document? When do we bump the minor and the major version numbers?

Should it be [Semantic Versioning](https://semver.org/) ?

File: openapi.yaml [hunk 2]
```diff
@@ -17,13 +17,30 @@ info:
...
     generated based on a **Project ID** and **Project API Key** that are found in the section 
     **Developers** on the [console](https://console.up42.com/). This token can only be used for a specific project.
     
     **Note**: We advise to safely store the Project ID and Project API Key and never share them with other users, because these credentials allow anyone to access your project and consume the UP42 credits associated with your account.
+    
+    # API Usage Constraints
+
+    Our mission is to optimize the access and usage of the diverse geospatial datasets and algorithms available on our marketplace. We expect that the UP42 API is used in a responsible and effective manner. 
+    
+    To prevent API abuse, we set up API usage constraints to guarantee the integrity and reliability of our service for all customers.
```
Comment (dc-up42) [inline openapi.yaml:32]: Where does text like this come from? Is anybody proofreading?

File: openapi.yaml (hunk 1 above)
Comment (dc-up42) [reply openapi.yaml:4]: Also, we are not changing the API itself here. Maybe we don't need to bump the version at all.

File: openapi.yaml [hunk 3]
```diff
@@ -17,13 +17,30 @@ info:
...
+    
+    ## Rate limiting
+    Based on the client IP address, we limit the number of HTTP requests that can be made with our API for a given time.
+    
+    **Current limit: 100 requests/second** for extended periods (including the possibility to burst up to 500 requests/second in a very short period).
+    Once the limit is exceeded, the API returns a **429 Too Many Requests** response status code.
+    
+    If you ever experience such an error code, please do one of the following:
+    - Adjust the request rate from your UP42 API client application such that the request rate follows an exponential backoff growth rate.
+    - Contact the customer support team and request the removal of this restriction, while providing a justification for this action. The customer support team will analyze the requests on a case-by-case basis and can waive this restriction for paying customers who provided a proper justification.
```
Comment (dc-up42) [inline openapi.yaml:42]: > will analyze the requests on a case-by-case basis and can waive this restriction for paying customers who provided a proper justification

Adding "for paying" here sounds a bit rough.

Also, do we really have a technical solution to change rate limits for a given customer?

File: openapi.yaml [hunk 4]
```diff
@@ -17,13 +17,30 @@ info:
...
+
+    Our mission is to optimize the access and usage of the diverse geospatial datasets and algorithms available on our marketplace. We expect that the UP42 API is used in a responsible and effective manner. 
+    
+    To prevent API abuse, we set up API usage constraints to guarantee the integrity and reliability of our service for all customers.
+    
+    ## Rate limiting
+    Based on the client IP address, we limit the number of HTTP requests that can be made with our API for a given time.
+    
+    **Current limit: 100 requests/second** for extended periods (including the possibility to burst up to 500 requests/second in a very short period).
+    Once the limit is exceeded, the API returns a **429 Too Many Requests** response status code.
```
Comment (dc-up42) [inline openapi.yaml:38]: A side note: this is highly coupled to our [infrastructure definition code](https://github.com/up42/infrastructure/blob/master/terraform/providers/kubernetes/manifests/application/ingress-nginx/05_ingress_api.yml#L12). 

We need to make sure that changing the rate limiting config in the infra repo will always be consistent with this. @bobthebuilderberlin @via-justa

File: openapi.yaml (hunk 2 above)
Comment (teoEnache) [reply openapi.yaml:32]: The text is transferred from the old docs. It was written by Antonio 2 years ago.

File: openapi.yaml (hunk 3 above)
Comment (teoEnache) [reply openapi.yaml:42]: For any topics related to rate limits, please ask Antonio. As mentioned, I did not write this article.

File: openapi.yaml (hunk 1 above)
Comment (teoEnache) [reply openapi.yaml:4]: Understood.

File: openapi.yaml (hunk 3 above)
Comment (dc-up42) [reply openapi.yaml:42]: @perusio , could you please comment?

Comment (teoEnache) [reply openapi.yaml:42]: Antonio just informed me that the content originates from Robert.

Comment (dc-up42) [reply openapi.yaml:42]: @bobthebuilderberlin , could you please comment? :D

File: openapi.yaml (hunk 4 above)
Comment (teoEnache) [reply openapi.yaml:38]: We can always review the content after the docs hub launch.

File: openapi.yaml (hunk 3 above)
Comment (teoEnache) [reply openapi.yaml:42]: we can revive the topic after the docs hub launch.

**Review by dc-up42 — APPROVED** [review]

File: openapi.yaml (hunk 4 above)
Comment (bobthebuilderberlin) [reply openapi.yaml:38]: @dc-up42 : Updating the infra repo with a comment:
https://github.com/up42/infrastructure/pull/494/files
Please review.

**Pull request merged**